lib/srv/regular/sshserver.go

```diff
@@ -990,11 +991,20 @@ func (s *Server) getRole() types.SystemRole {
 // getStaticLabels gets the labels that the server should present as static,
 // which includes EC2 labels if available.
 func (s *Server) getStaticLabels() map[string]string {
-	if s.cloudLabels == nil {
-		return s.labels
+	labels := make(map[string]string, len(s.labels))
+	if s.cloudLabels != nil {
+		for k, v := range s.cloudLabels.Get() {
+			labels[k] = v
+		}
 	}
-	labels := s.cloudLabels.Get()
-	// Let static labels override ec2 labels if they conflict.
+	// Let labels sent over ics override labels from instance metadata.
+	if s.inventoryHandle != nil {
+		for k, v := range s.inventoryHandle.GetUpstreamLabels(proto.LabelUpdateKind_SSHServer) {
+			labels[k] = v
+		}
```

Comment:
We need different rules for different sub-kinds so that we always have well-defined precedence/ordering.  I'd suggest renaming either renaming `LabelUpdateKind_SSHServer` to something like `LabelUpdateKind_SSHServerCouldLabels`, or adding an additional sub-kind to the label update message so that we can do something like:

```golang
s.inventoryHandle.GetUpstreamLabels(proto.LabelUpdateKind_SSHServer, types.SubKindCloudLabels)
```

That way if we add a different sub-kind later, we can ensure that they interact with other labels in a well-defined way and can have their own rules (e.g. if we ever support user-defined label overrides should probably support expiry/deletion, which isn't something we have for cloud labels).